spec/root_spec.cr

```diff
@@ -23,6 +22,24 @@ module PlaceOS::Api
         response.commit.should eq BUILD_COMMIT
       end
 
+      it "constructs service versions" do
+        WebMock.allow_net_connect = false
+        WebMock
+          .stub(:get, "triggers:3000/api/triggers/v2/version").to_return(body: %({"service":"triggers", "commit":"DEV", "version":"1", "build_time":"Tue Jun 01 01:00:00 UTC 2021", "platform_version":"DEV"}))
+        WebMock
+          .stub(:get, "127.0.0.1:3000/api/frontends/v1/version").to_return(body: %({"service":"frontends", "commit":"DEV", "version":"1", "build_time":"Tue Jun 01 01:00:00 UTC 2021", "platform_version":"DEV"}))
+        WebMock
+          .stub(:get, "127.0.0.1:3000/api/core/v1/version").to_return(body: %({"service":"core", "commit":"DEV", "version":"1", "build_time":"Tue Jun 01 01:00:00 UTC 2021", "platform_version":"DEV"}))
+        WebMock
+          .stub(:get, "rubber-soul:3000/api/rubber-soul/v1/version").to_return(body: %({"service":"rubber-soul", "commit":"DEV", "version":"1", "build_time":"Tue Jun 01 01:00:00 UTC 2021", "platform_version":"DEV"}))
+        WebMock
+          .stub(:get, "127.0.0.1:3000/api/server/version").to_return(body: %({"service":"dispatch", "commit":"DEV", "version":"1", "build_time":"Tue Jun 01 01:00:00 UTC 2021", "platform_version":"DEV"}))
+        version = Root.construct_versions
+        version.size.should eq(5)
+        version.first.service.should eq("frontends")
+        version.last.service.should eq("dispatch")
```

Comment:
There's no guarantee on the order of the asynchronous requests. 
You could check the `service` against `Root::SERVICES`

along the lines of
```crystal
versions.map(&.service).sort.should eq Root::SERVICES.sort
```